packages/ripple-binary-codec/src/types/xchain-bridge.ts

```diff
@@ -0,0 +1,128 @@
+import { BinaryParser } from '../serdes/binary-parser'
+
+import { AccountID } from './account-id'
+import { JsonObject, SerializedType } from './serialized-type'
+import { Buffer } from 'buffer/'
+import { Issue, IssueObject } from './issue'
+
+/**
+ * Interface for JSON objects that represent cross-chain bridges
+ */
+interface XChainBridgeObject extends JsonObject {
+  LockingChainDoor: string
+  LockingChainIssue: IssueObject | string
+  IssuingChainDoor: string
+  IssuingChainIssue: IssueObject | string
+}
+
+/**
+ * Type guard for XChainBridgeObject
+ */
+function isXChainBridgeObject(arg): arg is XChainBridgeObject {
+  const keys = Object.keys(arg).sort()
+  return (
+    keys.length === 4 &&
+    keys[0] === 'IssuingChainDoor' &&
+    keys[1] === 'IssuingChainIssue' &&
+    keys[2] === 'LockingChainDoor' &&
+    keys[3] === 'LockingChainIssue'
+  )
+}
+
+/**
+ * Class for serializing/deserializing XChainBridges
+ */
+class XChainBridge extends SerializedType {
+  static readonly ZERO_XCHAIN_BRIDGE: XChainBridge = new XChainBridge(
+    Buffer.concat([
+      Buffer.from([0x14]),
+      Buffer.alloc(40),
+      Buffer.from([0x14]),
+      Buffer.alloc(40),
+    ]),
+  )
+
+  static readonly TYPE_ORDER: { name: string; type: typeof SerializedType }[] =
+    [
+      { name: 'LockingChainDoor', type: AccountID },
+      { name: 'LockingChainIssue', type: Issue },
+      { name: 'IssuingChainDoor', type: AccountID },
+      { name: 'IssuingChainIssue', type: Issue },
+    ]
+
+  constructor(bytes: Buffer) {
+    super(bytes ?? XChainBridge.ZERO_XCHAIN_BRIDGE.bytes)
+  }
+
+  /**
+   * Construct a cross-chain bridge from a JSON
+   *
+   * @param value XChainBridge or JSON to parse into a XChainBridge
+   * @returns A XChainBridge object
+   */
+  static from<T extends XChainBridge | XChainBridgeObject>(
+    value: T,
+  ): XChainBridge {
+    if (value instanceof XChainBridge) {
+      return value
+    }
+
+    if (isXChainBridgeObject(value)) {
+      const bytes: Array<Buffer> = []
+      this.TYPE_ORDER.forEach((item) => {
+        const { name, type } = item
+        if (type === AccountID) {
+          bytes.push(Buffer.from([0x14]))
+        }
+        const object = type.from(value[name])
+        bytes.push(object.toBytes())
+      })
+      return new XChainBridge(Buffer.concat(bytes))
+    }
+
+    throw new Error('Invalid type to construct a XChainBridge')
```

Comment:
nit: Out of curiosity, is it `a XChain...` or is it `an XChain`? (I feel like `an`, but as long as it's consistent I can see it either way hahaha)

Reply:
I agree that it should be `an XChain` - some of my code was programmatically generated and I didn't add any a/an checks tbh